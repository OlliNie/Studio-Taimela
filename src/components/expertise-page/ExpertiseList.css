.li{
  display:flex;
  flex-direction: column;
  text-align: center;
  align-items: center;
  color:white;
  margin-bottom:3rem;
  p{ 
    width: 160px;
  }
  @media (min-width: 750px){
    margin-bottom:2rem;
    width: 34vw;
    flex-wrap:wrap;
  }
  @media(min-width:1100px){
    width:110px;
  }
}
.iconUl{
  padding:0;
  color: #262626;
  margin-top:1rem;
}
.ul{
  padding-left:0;
  flex-direction: column;
  display:flex;
  align-items: center;
  justify-content: space-around;
  margin-bottom:10rem;

  @media (min-width: 750px){
  flex-grow:1;
  flex-direction: row;
  padding:2rem 0 15rem 0;
  margin:0;
  flex-wrap: wrap;
  li:nth-child(odd){
    margin-left:auto;
  }
  li:nth-child(even){
    margin-right:auto;
  }
  li:nth-child(5){
    margin-right:auto;
    margin-left:auto;
  }
  }
  @media(min-width:900px){
    justify-content: center;
    padding:0;
  }
  @media (min-width: 1100px){
    align-self: center;
    width:800px;
    justify-content: space-between;
    padding:0;
    margin-top:2rem;
    li{
      margin:0;
    }
    li:nth-child(odd){
      margin-left:0;
    }
    li:nth-child(even){
      margin-right:0;
    }
    li:nth-child(5){
      margin-right:0;
      margin-left:0;
    }
  }
  @media (min-width: 1000px) and (min-height:400px){
    margin-bottom:37vh;
  }
}


