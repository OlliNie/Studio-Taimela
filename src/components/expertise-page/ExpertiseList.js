import React from 'react';
import styles from './ExpertiseList.css';
import { expertise } from '../../text/expertise';

export default function ExpertiseList() {
  const expertiseList = expertise.map((expertiseItem, i) => (
    <li key={i}>
      <section className={styles.li}>
        {expertiseItem.icon}
        <ul className={styles.iconUl}>
          {expertiseItem.text.map((text, textIndex) => (
            <li key={textIndex}>{text}</li>
          ))}
        </ul>
      </section>
    </li>
  ));
  return <ul className={styles.ul}>{expertiseList}</ul>;
}
