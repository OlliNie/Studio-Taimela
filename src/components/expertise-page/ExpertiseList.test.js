import React from 'react';
import { shallow } from 'enzyme';
import ExpertiseList from './ExpertiseList';

describe('ExpertiseList Component', () => {
  it('renders expertise list', () => {
    const wrapper = shallow(<ExpertiseList />);
    expect(wrapper).toMatchSnapshot();
  });
});
