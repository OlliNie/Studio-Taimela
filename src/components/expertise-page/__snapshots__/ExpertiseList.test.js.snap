// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpertiseList Component renders expertise list 1`] = `
<ul
  className="ul"
>
  <li
    key="0"
  >
    <section
      className="li"
    >
      <FaRandom
        size={36}
      />
      <ul
        className="iconUl"
      >
        <li
          key="0"
        >
          Able to quickly
        </li>
        <li
          key="1"
        >
          grasp complex,
        </li>
        <li
          key="2"
        >
          technical
        </li>
        <li
          key="3"
        >
          solutions
        </li>
      </ul>
    </section>
  </li>
  <li
    key="1"
  >
    <section
      className="li"
    >
      <FaGlobeAmericas
        size={36}
      />
      <ul
        className="iconUl"
      >
        <li
          key="0"
        >
          Extensive
        </li>
        <li
          key="1"
        >
          experience in
        </li>
        <li
          key="2"
        >
          international
        </li>
        <li
          key="3"
        >
          business
        </li>
      </ul>
    </section>
  </li>
  <li
    key="2"
  >
    <section
      className="li"
    >
      <FaComments
        size={36}
      />
      <ul
        className="iconUl"
      >
        <li
          key="0"
        >
          Translate
        </li>
        <li
          key="1"
        >
          tech talk into
        </li>
        <li
          key="2"
        >
          non-technical
        </li>
        <li
          key="3"
        >
          language
        </li>
      </ul>
    </section>
  </li>
  <li
    key="3"
  >
    <section
      className="li"
    >
      <FaStar
        size={36}
      />
      <ul
        className="iconUl"
      >
        <li
          key="0"
        >
          Can see past
        </li>
        <li
          key="1"
        >
          complexity to
        </li>
        <li
          key="2"
        >
          bring out the
        </li>
        <li
          key="3"
        >
          best
        </li>
      </ul>
    </section>
  </li>
  <li
    key="4"
  >
    <section
      className="li"
    >
      <FaFileAlt
        size={36}
      />
      <ul
        className="iconUl"
      >
        <li
          key="0"
        >
          Both
        </li>
        <li
          key="1"
        >
          copywriting
        </li>
        <li
          key="2"
        >
          and layout
        </li>
        <li
          key="3"
        >
          design
        </li>
      </ul>
    </section>
  </li>
</ul>
`;
